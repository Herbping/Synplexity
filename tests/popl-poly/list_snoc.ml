type 'a list =
  | Nil
  | Cons of 'a * 'a list

let rec list_snoc : 'a -> 'a list -> 'a list |>
/\(a -> /\([ ] -> [a], ['a] -> ['a; a], ['a; 'a] -> ['a; 'a; a])) = ?
