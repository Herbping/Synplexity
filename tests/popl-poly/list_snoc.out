(* Generated Thu Oct  8 20:27:09 2015 *)
fix f1 (p1:'a) : 'a list -> 'a list ->
  fix f2 (l1:'a list) : 'a list ->
    match l1 with
    | Nil u1 -> Cons (p1, Nil)
    | Cons t1 -> Cons (p1, f2 (#2 t1))
